### PR TITLE
Protocol Definitions Typing Improvements

### PR DESCRIPTION
**Pull request by anthony-murphy**
## Description

 - Change usages of `any` to `unknown`. This changes nothing a runtime, but forces typescript users to safely access the value
 - Add `null` to ISequencedDocumentMessage.client id, as `null` is and always has been a valid value at runtime which must be handled. Server generated message currently have `null` as the value for `clientId`.
 - Change references to `op` or `delta` in comments to message, as messages do not map one to one with ops in many cases
 - Fix spelling errors.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
+	"cSpell.words": [
+		"endregion",
+		"injective",
+		"losslessly",
+		"mitigations",
+		"multinomial",
+		"nacked",
+		"nonfinite",
+		"pseudorandomly",
+		"retriable",
```
Comment (anthony-murphy) [inline .vscode/settings.json:35]: surveyed the code, retriable is more commonly used than retryable, so going with that, even typing this retryable gets re-underline via spell check, but retriable does not, so seems like the correct choice

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
-        "unaugmented"
-    ],
+	// Custom dictionary for 'streetsidesoftware.code-spell-checker' extension.
+	"cSpell.words": [
+		"endregion",
+		"injective",
+		"losslessly",
+		"mitigations",
+		"multinomial",
+		"nacked",
```
Comment (anthony-murphy) [inline .vscode/settings.json:32]: also added nacked.

File: common/lib/protocol-definitions/api-report/protocol-definitions.api.md [hunk 3]
```diff
@@ -282,20 +282,20 @@ export interface ISequencedDocumentAugmentedMessage extends ISequencedDocumentMe
 
 // @public
 export interface ISequencedDocumentMessage {
-    clientId: string;
+    clientId: string | null;
```
Comment (anthony-murphy) [inline common/lib/protocol-definitions/api-report/protocol-definitions.api.md:285]: we've been putting this off for a long while, but it shouldn't be that bad to consume, as the code needs to handle it already.

**Comment by anthony-murphy** [comment]
Is this the correct branch (main) to target for breaking changes in our common packages? The will need to flow through server to reach client, which i know is only modified in main.

**Comment by tylerbutler** [comment]
> Is this the correct branch (main) to target for breaking changes in our common packages? The will need to flow through server to reach client, which i know is only modified in main.

Yes, it is. There have already been breaking changes to protocol-defs, and it's bumped to 2.0 on main in preparation. @ChumpChief and @ghoshkaj were the folks who made other changes. When you are ready to integrate into server and need a release, let me know.

File: common/lib/protocol-definitions/api-report/protocol-definitions.api.md (hunk 3 above)
Comment (tylerbutler) [reply common/lib/protocol-definitions/api-report/protocol-definitions.api.md:285]: Is `null` correct? Just double-checking.

File: common/lib/protocol-definitions/src/protocol.ts [hunk 4]
```diff
@@ -212,9 +206,11 @@ export interface IBranchOrigin {
  */
 export interface ISequencedDocumentMessage {
 	/**
-	 * The client ID that submitted the delta.
+	 * The client ID that submitted the message.
+	 * For server generated messages the clientId will be null;
 	 */
-	clientId: string;
+	// eslint-disable-next-line @rushstack/no-new-null
+	clientId: string | null;
```
Comment (tylerbutler) [inline common/lib/protocol-definitions/src/protocol.ts:213]: The lint suppression answers my question about whether this was intentional, but why is null correct here?

**Review by tylerbutler — APPROVED** [review]
Since this is an independent package, go ahead and add a note to the CHANGELOG.md. It can be done later, too, so it's not a blocker.

File: common/lib/protocol-definitions/api-report/protocol-definitions.api.md (hunk 3 above)
Comment (anthony-murphy) [reply common/lib/protocol-definitions/api-report/protocol-definitions.api.md:285]: yes. it is. its what the server currently sends over the wire, the current typing is wrong to not include null.

**Review by ChumpChief — APPROVED** [review]
Love it!

**Review by alexvy86 — COMMENTED** [review]
Should this PR include a major bump in protocol-definitions? I thought one of the reasons we were holding on this change is that it's breaking.

**Comment by ChumpChief** [comment]
> Should this PR include a major bump in protocol-definitions? I thought one of the reasons we were holding on this change is that it's breaking.

I bumped it in my #15166 and we haven't released since then, so 2.0.0 is still the correct next major to release.

**Pull request merged**